### DISCUSSION TITLE
Provide a built-in updater that is shipped with the software itself

### DISCUSSION THREAD

**Discussion by ghost** [rustdesk/rustdesk#4600] [Feature Request]
### Bug Description

When the software is installed and a new version is released, it will not update to a new version.

### How to Reproduce

Install an out-dated version and it will remain on that version when a new version is released.

### Expected Behavior

It is expected for software to provide the necessary built-in means to make updating itself easy.

### Operating system(s) on local side and remote side

Any operating system that does not provide software repositories.

### RustDesk Version(s) on local side and remote side

Any.

### Screenshots

![no-automatic-update](https://github.com/rustdesk/rustdesk/assets/97630012/5f05694d-79e1-446c-b96b-16fab51b2334)

### Additional Context

The community has provided third party scripts to perform this job, but the topic is about a shipped solution within the software itself. Linking to these scripts provides no relevance.

**Comment by test2a** [comment]
yes. on windows at least, there is no way to keep the software up to date.

on linux, we can do a simple thing

https://assafmo.github.io/2019/05/02/ppa-repo-hosted-on-github.html

i suppose linux and mac would benefit from this for auto update. 

windows is a bit difficult but rustdesk can always use chocolaty and other tools for example but i suppose windows would need internal updater anyway

**Comment by schtritoff** [comment]
On Windows, you can install it via WinGet https://winstall.app/apps/RustDesk.RustDesk and after installation you can use something like this auto updater for winget compatible apps https://github.com/Romanitho/Winget-AutoUpdate

**Reply by ghost** [reply]
Please see the additional context section in the original post.

We are not here to discuss third party solutions. We are here to discuss a built-in solution.

**Reply by remoover** [reply]
Please do the auto-update as a built-in solution; we started using Rustdesk on over 50+ computers and can't figure out an excellent way to auto-update... Many thanks for your good job!
